### PR TITLE
[FX-5545] Migrate Note to TailwindCSS

### PR DESCRIPTION
**Pull request by augustobmoura**
[FX-5545]

### Description

Migrate Note to TailwindCSS

### How to test

- [Temploy](https://picasso.toptal.net/fx-5545-note-to-tailwind)
- Check Happo and Storybook

### Screenshots

| Before.                                 | After.                                  |
| --------------------------------------- | --------------------------------------- |
| Insert screenshots or screen recordings | Insert screenshots or screen recordings |

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [x] Make sure that additions and changes on the design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [x] Annotate all `props` in component with documentation
- [x] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- [x] Covered with tests ([visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md) included)

**Breaking change**

- [x] codemod is created and showcased in the changeset
- [x] test alpha package of Picasso in StaffPortal

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-5545]: https://toptal-core.atlassian.net/browse/FX-5545?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .changeset/little-papayas-eat.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'@toptal/picasso-note': minor
+'@toptal/picasso': minor
```
Comment (TomasSlama) [inline .changeset/little-papayas-eat.md:3]: why minor? 🤔 we don't introduce anything new for the user, right?

as we added tailwind-merge in peer dependencies we should probably mark this as breaking change 🤔

File: packages/base/Note/src/Note/Note.tsx [hunk 2]
```diff
@@ -1,25 +1,25 @@
...
 ) {
-  const classes = useStyles()
-
   return (
-    <div ref={ref} className={cx(classes.root, className)} {...rest}>
+    <div
+      ref={ref}
+      className={twMerge(
+        `rounded-md border border-solid border-gray-200 py-6 pr-6 pl-[28px] relative overflow-hidden bg-white
+        before:content-[''] before:bg-yellow-500 before:block before:h-full before:absolute before:top-0 before:left-0 before:w-[4px]`,
```
Comment (TomasSlama) [inline packages/base/Note/src/Note/Note.tsx:18]:
```suggestion
        'before:content-['']',
        'before:bg-yellow-500',
        'before:block',
        'before:h-full',
        'before:absolute',
        'before:top-0',
        'before:left-0',
        'before:w-[4px]',
```

NIT: personally I find it more readable when classes with the same prefix are on new line as my eyes can ignore the prefix and actually read the styles

**Review by TomasSlama — APPROVED** [review]

**Review by dmaklygin — APPROVED** [review]
Checked the code. Looks good!

File: .changeset/little-papayas-eat.md (hunk 1 above)
Comment (augustobmoura) [reply .changeset/little-papayas-eat.md:3]: Makes sense, I will update the changeset message and mark note as a major bump, main picasso package will be bumped as a patch

**Pull request merged**